tests/driver/index.ts

```diff
@@ -0,0 +1,438 @@
+          return def["known_good_version"] ?? "";
+        }
+      }
+      return "";
+    } else {
+      return this.inputArgs.linterVersion;
+    }
+  };
+
+  // Setup a sandbox test directory by copying in test contents and conditionally:
+  // 1. Creating a git repo
+  // 2. Dumping a newly generated trunk.yaml
+  // 3. Enabling the specified 'linter'
+  async SetUp() {
+    this.sandboxPath = fs.mkdtempSync(path.join(os.tmpdir(), TEMP_PREFIX));
+    if (this.sandboxPath) {
+      // Create repo
+      const sourceDir = path.join(this.testDir, "..");
+      fs.cpSync(sourceDir, this.sandboxPath, { recursive: true });
+
+      if (this.setupSettings.setupGit) {
+        this.gitDriver = git.simpleGit(this.sandboxPath);
+        this.gitDriver.init();
+      }
+
+      if (this.setupSettings.setupTrunk) {
+        // Initialize trunk via config
+        fs.mkdirSync(path.join(this.sandboxPath, ".trunk"));
+        fs.writeFileSync(
+          path.join(this.sandboxPath, ".trunk/trunk.yaml"),
+          this.TrunkYamlContents()
+        );
+      }
+
+      // Enable tested linter if specified
+      if (this.linter) {
+        const daemonCommand = `${this.inputArgs.cliPath ?? "trunk"} daemon launch --monitor=false`;
+        exec(daemonCommand, { cwd: this.sandboxPath, timeout: ENABLE_TIMEOUT });
+        try {
+          const version = this.ExtractLinterVersion();
+          const versionString = version.length > 0 ? `@${version}` : "";
+          const linterVersionString = `${versionString}${this.linter}`;
+
+          // Prefer calling enable over editing trunk.yaml directly because it also handles runtimes, etc.
+          const enableCommand = `${
+            this.inputArgs.cliPath ?? "trunk"
+          } check enable ${linterVersionString} --monitor=false`;
+          execSync(enableCommand, { cwd: this.sandboxPath });
+        } catch (error) {
+          console.warn(`Failed to enable ${this.linter} with message ${(error as Error).message}`);
+        }
+      }
+    }
+  }
+
+  // Delete the sandbox testing directory and its contents, as well as removing any trunk information
+  // associated with it in order to prune the cache.
+  async TearDown() {
+    const deinitCommand = `${this.inputArgs.cliPath ?? "trunk"} deinit`;
+    exec(deinitCommand, { cwd: this.sandboxPath, timeout: 1000 });
+    if (this.sandboxPath) {
+      fs.rmSync(this.sandboxPath, { recursive: true });
+    }
+  }
+
+  // Run a specified trunk command with 'args' (omit 'trunk') and additional options.
+  async Run(args: string, execOptions?: ExecOptions) {
+    const fullCommand = `${this.inputArgs.cliPath ?? "trunk"} ${args}`;
+    return execpromise(fullCommand, { cwd: this.sandboxPath, ...execOptions });
+  }
+
+  // Run a 'trunk check' command with additional options.
+  async RunCheck(
+    args = "",
+    targetAbsPath?: string,
+    resultJsonPath: string = path.join(this.sandboxPath ?? "", "result.json")
+  ) {
```

Comment:
Methods / functions with many arguments are better suited to used named-style arguments using objects—it helps make code more more clear. I recommend it for almost all methods / functions. Example:

```suggestion
  async RunCheck({
    args = "",
    targetAbsPath,
    resultJsonPath = path.join(this.sandboxPath ?? "", "result.json"),
  }: {
    args?: string,
    targetAbsPath?: string,
    resultJsonPath?: string,
  }) {
```

this way something could call `RunCheck` like this
```ts
RunCheck({
  args: "",
  targetAbsPath: __dirname,
  resultJsonPath: __dirname,
})
```